spring/spring-data/src/test/java/io/quarkus/ts/spring/data/rest/additional/OpenAPIIT.java

```diff
@@ -61,12 +61,11 @@ void rolesAllowedResourceAuthPermitted() {
         assertNotNull(json.getString("paths.\"/article-jpa/{id}\".get"));
 
         json.setRootPath("");
-        assertEquals("admin", json.getString("paths.\"/secured/deny-all/{id}\".get.security[0].SecurityScheme[0]"));
-        assertEquals("admin", json.getString("paths./secured/roles-allowed.get.security[0].SecurityScheme[0]"));
-        assertEquals("admin", json.getString("paths.\"/secured/roles-allowed/{id}\".get.security[0].SecurityScheme[0]"));
-
-        // TODO: https://github.com/quarkusio/quarkus/issues/30997
-        //        assertEquals("user", json.getString("paths.\"/secured/roles-allowed/{id}\".delete.security[0].SecurityScheme[0]"));
+        // non-Oauth2 security requirement object should be empty
```

Comment:
Can we enable Oauth2 for Spring-data? Now it looks like there shouldn't be any security for spring and it doesn't seem right.

Reply:
- In that case we won't have testing non-oauth2 
- It doesn't look like there is any security for Spring as OpenAPI definition says there is basic auth (situation didn't change in that respect), please see here https://github.com/quarkus-qe/quarkus-test-suite/blob/main/spring/spring-data/src/test/java/io/quarkus/ts/spring/data/rest/additional/OpenAPIIT.java#L84 only thing that changed is that path security scheme doesn't contain security requirement object as the specification doesn't allow that
- If I enable it for Spring Data, it will cost addition resources\
- OAuth2 OpenAPI definition is tested by OIDC tests I added (sits on top of Oauth2); OpenAPI generation is not really related to Spring/RESTEasy Reactive etc. The information is based on the security information provider (so here it is provided by vertx-http, for OIDC it is oidc extension) that provides the information to the OpenAPI extension

@fedinskiy I'd rather not do it unless I understand pros.

Reply:
Ok, so it is expected result of fixing https://github.com/quarkusio/quarkus/issues/27373.
I suggest to change the wording of the comment to make it a little bit more clear.
Something like "OpenAPI should generate per-scope security requirements only for OAuth2". Or post a link to https://github.com/OAI/OpenAPI-Specification/blob/3.0.1/versions/3.0.1.md#patterned-fields-3. Or both :)

Reply:
Ok, so it is expected result of fixing https://github.com/quarkusio/quarkus/issues/27373.
I suggest to change the wording of the comment to make it a little bit more clear.
Something like "OpenAPI should generate per-scope security requirements only for OAuth2". Or post a link to https://github.com/OAI/OpenAPI-Specification/blob/3.0.1/versions/3.0.1.md#patterned-fields-3. Or both :)